src/cli/node/json_rpc_client.rs

```diff
@@ -978,8 +977,11 @@ pub fn send_st(
     let (dry, _): (BuildStakeResponse, _) =
         issue_method("stake", Some(params), &mut stream, id.next())?;
 
-    let validator_address = validator
-        .try_do_magic(|hex_str| PublicKeyHash::from_bech32(get_environment(), &hex_str))?;
+    let validator_address = {
+        let pkh_bytes =
+            hex::decode(authorization.clone().chars().take(40).collect::<String>()).unwrap();
+        PublicKeyHash::from_bytes(pkh_bytes.as_slice())?
+    };
     if validator_address != dry.validator {
         bail!(
             "The specified validator ({}) does not match the validator recovered from the authorization string ({}), please double check all arguments.",
```

Comment:
This message becomes a bit weird with the validator being recovered from the authorization string.

If you specify an incorrect withdrawer, the dry run will result in that withdrawer being recovered but with a validator that does not match the validator in the authorization string.

My proposal would be to update this message:
```rust
bail!(
    "The validator recovered from the authorization string ({}) does not match the validator calculated using the specified withdrawer ({}). Please double check you are supplying the same withdrawer as the one used to generate the authorization string.",
    validator_address,
    dry.validator.to_string(),
);
```